entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EMController.java

```diff
@@ -851,24 +849,17 @@ private ResponseEntity<String> checkExistingEntity(
   private ResponseEntity<String> redirectDeprecated(
       EntityRecord deprecatedEntity, EntityTypes type, String identifier)
       throws EntityRemovedException {
-    // check if there is an entity in the sameAs ref links starting with "http://data.europeana.eu/"
-    // and so, redirect to it
-    List<String> sameAsRefLinks = deprecatedEntity.getEntity().getSameReferenceLinks();
-    if (sameAsRefLinks != null) {
-      Optional<String> sameAsLink =
-          sameAsRefLinks.stream()
-              .filter(el -> el.contains(WebEntityFields.BASE_DATA_EUROPEANA_URI))
-              .findFirst();
-      if (sameAsLink.isPresent()) {
-        // return 301 redirect
-        return ResponseEntity.status(HttpStatus.MOVED_PERMANENTLY)
-            .location(URI.create(sameAsLink.get()))
-            .build();
-      }
+    //check if there is any enabled entity in the db, containing the entityId of this entity
+    //in the sameAs ref links and redirect to it
+    Optional<EntityRecord> entityRedirect = entityRecordService.findEntityDupplicationBySameAs(deprecatedEntity.getEntityId());
```

Comment:
the deprecatedEntity.getEntity().getSameReferenceLinks() needs to be used to cover the Concepts as well.
All data.europeana.eu URIs needs to be used for